models/content.rb

```diff
@@ -101,6 +101,6 @@ def self.summary what
 
   def set_username
     # avoid having to look this attribute up later, since it does not change
-    self.author_username = author.username
+    self.author_username = self.retired_username ? self.retired_username : author.username
```

Comment:
For the contents collection, the "author_username" field gets set on save each time. I added a separate "retired_username" field to override author.username because of this.